nav2_waypoint_follower/include/nav2_waypoint_follower/waypoint_follower.hpp

```diff
@@ -133,10 +139,13 @@ class WaypointFollower : public nav2_util::LifecycleNode
   rclcpp::CallbackGroup::SharedPtr callback_group_;
   rclcpp::executors::SingleThreadedExecutor callback_group_executor_;
   std::shared_future<rclcpp_action::ClientGoalHandle<ClientT>::SharedPtr> future_goal_handle_;
+  int error_code_;
```

Comment:
remove?

Reply:
Oh this is used to store the current one, maybe `current_error_code_`?

Reply:
Better yet, a struct containing the values of `current_goal_status_` and `error_code_` since they're both populated in the `resultCallback`, which itself would be named `current_goal_status_` with the int error code and the ActionStatus enum.